### DISCUSSION TITLE
Make type "InvokeCallback" public or offer an event type for "sendBack"

### DISCUSSION THREAD

**Discussion by phcoliveira** [statelyai/xstate#5053] [Ideas]
The title may be confusing, but the issue is rather simple.

Previously, the type `InvokeCallback` was exported by XState. It wasn't public per se, as I guess it was never meant to be used by consumers of the XState library.

Using XState version 5.15, I could import such type and wrap the function `fromCallback` with a proper typing for its `sendBack` function. I used to wrap it in a function named `enhancedFromCallback`. You can see a discussion about it here: https://github.com/statelyai/xstate/issues/4666#issuecomment-2102799308

```ts
import {fromCallback} from 'xstate';
import type {AnyEventObject, EventObject, NonReducibleUnknown} from 'xstate';
import type {InvokeCallback} from 'xstate/dist/declarations/src/actors/callback';

export function enhancedFromCallback<
	TReceive extends EventObject = AnyEventObject,
	TSend extends EventObject = AnyEventObject,
	TInput = NonReducibleUnknown
>(invokeCallback: InvokeCallback<TReceive, TSend, TInput>) {
	return fromCallback(invokeCallback);
}
```

With version 5.17, that is no longer possible. And I find very strange that a similar provision is now offered for `({ emit })`, but not for `({ sendBack })`, which is arguably much more useful.

So, I believe there must be a reason for not offering the means for typing `sendBack` out of the box, I just don't know what this reason might be. In any case, can we at least the `InvokeCallback` publicly available so I can continue my approach of using an `enhancedFromCallback`?

**Comment by davidkpiano** [comment]
https://github.com/statelyai/xstate/pull/5054
